tempesta_fw/http.c

```diff
@@ -1038,30 +996,13 @@ tfw_h2_send_resp(TfwHttpReq *req, int status, unsigned int stream_id)
 
 	/* Create and set frame header and set payload for DATA. */
 	if (body->data) {
-		frame_hdr.length = body->len;
-		frame_hdr.type = HTTP2_DATA;
-		frame_hdr.flags = HTTP2_F_END_STREAM;
-		tfw_h2_pack_frame_header(buf, &frame_hdr);
-
-		if (tfw_http_msg_expand_data(&resp->mit.iter, skb_head, &f_hdr,
-					     NULL)
-		    || tfw_http_msg_expand_data(&resp->mit.iter, skb_head, body,
-						NULL))
+		if (tfw_http_msg_expand_data(&resp->mit.iter, skb_head, body,
+					     NULL))
 			goto err_setup;
 	}
 
-	/*
-	 * Set the frame header for HEADERS. Note, that we leave the place for
-	 * it at the beginning of the response - due to @mit->start_off setting
-	 * above (before the first data is written into the target skb).
-	 */
-	frame_hdr.length = mit->acc_len;
-	frame_hdr.type = HTTP2_HEADERS;
-	frame_hdr.flags = HTTP2_F_END_HEADERS;
-	if (!body->data)
-		frame_hdr.flags |= HTTP2_F_END_STREAM;
-	tfw_h2_pack_frame_header(buf, &frame_hdr);
-	memcpy_fast((*skb_head)->data, buf, sizeof(buf));
+	if (tfw_h2_make_frames(resp, stream_id, mit->acc_len, true, false))
```

Comment:
Same here, `tfw_h2_make_frames()` always check `resp->body` and doesn't generate body frames if body absent. (HEADERS frame header also depends on that, because `STREAM_END` flag may only be set if no body is provided). 

Besides, user may override body with `response_body`, so static response will have some body.